plugins/modules/podman_network.py

```diff
@@ -386,7 +386,7 @@ def diffparam_opt(self):
             vlan_before = self.info['plugins'][0].get('vlan')
         except (IndexError, KeyError):
             vlan_before = None
-        vlan_after = self.params['opt'].get('vlan') if self.params['opt'] else None
+        vlan_after = self.params.get('opt', {}).get('vlan', None)
```

Comment:
It won't work that way. `self.params` are Ansible `params` which are always specified, and is `None` if not set. In `self.params.get('opt', {})` you'll always get `None` in case `opt` is not set in module args.
That's why it's implemented as implemented :) I can't understand how these changes are related to the issue, can you elaborate?

Reply:
I see. Let me please fix it in a different place: #435